.golangci.yml

```diff
@@ -33,6 +33,28 @@ linters-settings:
   govet:
     # report about shadowed variables
     check-shadowing: true
+    settings:
+      printf:
+        # We are overgenerous here to accomodate potential not yet existing functions.
```

Comment:
Oh we could probably trim this back. The plain Logger is a very limited subset by design, so that things like the official Zap logger implement it. All of the extra stuff was kept in SugaredLogger.

Reply:
Oh I was actually thinking fewer `Logger` methods, since some are only on `SugaredLogger`, but I may not be up to date with the API....

It looks like for `Logger` we currently have: 
- Debug
- Info
- Warn
- Error
- Panic
- Fatal

And for `SugaredLogger` we have:
- AssumptionViolation
- Critical
- Trace
Plus, it embeds `Logger`, so do we need to explicitly list out those method again in order to catch cases where someone calls a simple `Logger` method through a `SugaredLogger`?

Reply:
Fortunately, there is no need to repeat - see my other comment.